extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -62,10 +67,19 @@ public void build(ApplicationInfoBuildItem applicationInfo,
             throw new RuntimeException("Unable to setup environment for generating Kubernetes resources", e);
         }
 
+        ApplicationPropertiesConfigSource applicationProperties = new ApplicationPropertiesConfigSource.InJar();
```

Comment:
I don't think you should access the config source directly: use `ConfigProvider` instead.

Reply:
This extension is only meant to be used configured at build time (since it generated kubernetes manifests) so in this case it shouldn't make a difference.
However I agree that `ConfigProvider` would be the proper solution

Reply:
Switched to ConfigProvider.